bmc/inventory.go

```diff
@@ -0,0 +1,78 @@
+package bmc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/bmc-toolbox/bmclib/devices"
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+
+	bmclibErrs "github.com/bmc-toolbox/bmclib/errors"
+)
+
+// InventoryGetter defines methods to retrieve device hardware and firmware inventory
+type InventoryGetter interface {
+	GetInventory(ctx context.Context) (device *devices.Device, err error)
+}
+
+type inventoryGetterProvider struct {
+	name string
+	InventoryGetter
+}
+
+// GetInventory returns hardware and firmware inventory
+func GetInventory(ctx context.Context, generic []inventoryGetterProvider) (device *devices.Device, metadata Metadata, err error) {
```

Comment:
should `Get` be avoided here? -- [ref](https://go.dev/doc/effective_go#Getters)

Reply:
Thanks for a taking a look @jacobweinstock, updated in b5b5b13306.